tsrc/test/helpers/cli.py

```diff
@@ -27,14 +22,12 @@ def run_and_fail(self, *args: str) -> tsrc.Error:
         ui.info(ui.red, " (expecting failure)")
         with pytest.raises(tsrc.Error) as e:
             tsrc.cli.main.testable_main(args)
-        return e.value  # type: ignore
+        return e.value
 
-    def run_and_fail_with(self, error: ErrorCtor, *args: str) -> tsrc.Error:
-        ui.info(">", ui.bold, "tsrc", *args, end="")
-        ui.info(ui.red, " (expecting failure)")
-        with pytest.raises(error) as e:
-            tsrc.cli.main.testable_main(args)
-        return e.value  # type: ignore
+    def run_and_fail_with(self, error: type, *args: str) -> tsrc.Error:
```

Comment:
note for reviewers: type is the type of classes in Python

Reply:
Shouldn't `type` be `Any` (imported above) or `Exception`, the base class for any python error?

Reply:
That should work to

Reply:
Ah, right. I should have said `typing.Type[Exception]` which will let you accept any subclass of `Exception`: https://docs.python.org/3/library/typing.html#typing.Type

> a variable annotated with `Type[C]` may accept values that are classes themselves – ...
> The fact that `Type[C]` is covariant implies that all subclasses of `C` should implement the same constructor signature and class method signatures as `C`.

As it's better to be more explicit when possible, you don't want to say that you'll accept literally anything:

> `Type[Any]` is equivalent to `Type` which in turn is equivalent to `type`, which is the root of Python’s metaclass hierarchy.

Reply:
Oh I see. Thank you for your help!